cli/args.go

```diff
@@ -0,0 +1,121 @@
+package cli
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/docopt/docopt-go"
+)
+
+// Environment variable support:
+//   PYREVT_TELEMETRY_DB_CONNSTRING
+//   PYREVT_TELEMETRY_SCRIPTS_TABLE
+//   PYREVT_TELEMETRY_EVENTS_TABLE
+//   PYREVT_TELEMETRY_PORT
+//   PYREVT_TELEMETRY_HTTPS
+//   PYREVT_TELEMETRY_DEBUG
+//   PYREVT_TELEMETRY_TRACE
+//   PYREVT_TELEMETRY_EXENAME
+//   PYREVT_TELEMETRY_VERSION
+
+type Options struct {
+	ExeName      string `json:"exe_name"`
+	Version      string `json:"version"`
+	Opts         *docopt.Opts
+	ConnString   string `json:"connection_string"`
+	ScriptsTable string `json:"script_table"`
+	EventsTable  string `json:"events_table"`
+	Port         int    `json:"server_port"`
+	Https        bool   `json:"https"`
+	Debug        bool   `json:"debug_mode"`
+	Trace        bool   `json:"trace_mode"`
+}
+
+func getExeName() string {
+	return strings.TrimSuffix(
+		filepath.Base(os.Args[0]),
+		filepath.Ext(os.Args[0]),
+	)
+}
+
+func NewOptions() *Options {
+	argv := os.Args[1:]
+
+	parser := &docopt.Parser{
+		HelpHandler: printHelpAndExit,
+	}
+
+	opts, _ := parser.ParseArgs(help, argv, version)
+
+	connString, _ := opts.String("<db_uri>")
+	scriptTable, _ := opts.String("--scripts")
+	eventTable, _ := opts.String("--events")
+	port, _ := opts.Int("--port")
+	https, _ := opts.Bool("--https")
+
+	debug, _ := opts.Bool("--debug")
+	trace, _ := opts.Bool("--trace")
+
+	// Environment variable fallback
+	if connString == "" {
+		connString = os.Getenv("PYREVT_TELEMETRY_DB_CONNSTRING")
+	}
+	if scriptTable == "" {
+		scriptTable = os.Getenv("PYREVT_TELEMETRY_SCRIPTS_TABLE")
+	}
+	if eventTable == "" {
+		eventTable = os.Getenv("PYREVT_TELEMETRY_EVENTS_TABLE")
+	}
```

Comment:
This is another ranting for the future: do we really need to have dynamic table names? Especially now that you're introducing the APIs to get the data out of the DB....